app/controllers/spree/admin/products_controller.rb

```diff
@@ -160,10 +160,16 @@ def collection_actions
       private
 
       def product_set_from_params(params)
-        collection_hash = Hash[params[:products].each_with_index.map { |p, i| [i, p] }]
+        collection_hash = Hash[products_params.each_with_index.map { |p, i| [i, p] }]
         Spree::ProductSet.new(collection_attributes: collection_hash)
       end
 
+      def products_params
+        params.require(:products).map do |product|
+          ActionController::Parameters.new(product.to_hash).permit(:id, :name)
+        end
+      end
```

Comment:
Can't we do this?
```ruby
      def products_params
        params.require(:products).map do |product|
          product.permit(:id, :name)
        end
      end
```

Reply:
yes we can!!! I'll switch.
Thanks!

Reply:
Awesome, that looks a lot nicer ad is easier to read. :+1: